### ISSUE TITLE
Treat "container-title-short" the same as "journalAbbreviation"

### ISSUE THREAD

**Issue by rmzelle** [Juris-M/citeproc-js#84]
Hi Frank, @zuphilip ran into the following issue (https://github.com/citation-style-language/styles/pull/3618#discussion_r208234949):

```xml
  <bibliography>
    <layout>
      <group delimiter="/">
        <text variable="container-title-short"/>
        <text variable="container-title" form="short"/>
        <text variable="container-title"/>
      </group>
    </layout>
  </bibliography>
```

produces

> Comp. Polit. Stud./Comp. Polit. Stud./Comparative Political Studies
> TWO/The World is Open/The World is Open

The first line is a journal article, with

```
"container-title": "Comparative Political Studies",
"journalAbbreviation": "Comp. Polit. Stud."
```

So far, so good. Both `variable="container-title-short"` and `"container-title" form="short"` produce the abbreviated container-title. (see https://github.com/citation-style-language/csl-editor/blob/1319eeaebb5ca13f001d7773f8cc30a923718466/src/exampleData.js#L56 for CSL JSON)

However, for the second line, we have a chapter with:

```
"container-title": "The World is Open",
"container-title-short": "TWO"
```

Here, `"container-title" form="short"` gives the full title. Shouldn't it give the contents of "container-title-short"?

I was able to reproduce this in Zotero 5.0.54.

**Comment by zuphilip** [comment]
Is `journalAbbreviation` a typo in @rmzelle 's first post? Because it is a Zotero field and not part of CSL JSON as the remaining fields.

Currently in Zotero one would need to add `container-title-short` by the extra field, where `journalAbbreviation` can be entered directly and is more specific semantically. Thus, in cases where both fields are entered in Zotero, I would vote to prefer the `journalAbbreviation` field.

**Comment by fbennett** [comment]
``citeproc-js`` handles ``journalAbbreviation`` directly, which must be a legacy thing from our very early days. The difference in behavior between the examples is not due to the difference in item type (``article-journal`` vs ``chapter``), but the difference in variable name. The processor is currently just not picking up ``container-title-short`` as an equivalent to (legacy/non-CSL) ``journalAbbreviation``.

Thinking more about this, there are two layers to the issue (where Zotero or other app is concerned):
1. The app can control priority as between ``journalAbbreviation`` and ``container-title-short``, by passing only one of them when both are present. 
2. In the event that both are delivered, the processor needs to choose between them.

Regardless of behavior in the processor, Zotero can preserve the behavior that its users expect by writing its ``journalAbbreviation`` value onto ``container-title-short``. We can limit the issue here to what the processor *should* do, as a CSL implementation. If only ``container-title-short`` is officially a variable, would it make sense to prefer that, and fall back to ``journalAbbreviation`` only if it's not present?

CC @rmzelle @adam3smith

**Comment by rmzelle** [comment]
> Is journalAbbreviation a typo in @rmzelle 's first post? Because it is a Zotero field and not part of CSL JSON as the remaining fields.

No, it's part of the CSL JSON model (https://github.com/citation-style-language/schema/blob/v1.0.1/csl-data.json#L65). Before CSL introduced `title-short` and `container-title-short`, Frank had to store the short forms of the `title` and `container-title` in JSON somehow for citeproc-js, so he came up with `shortTitle` and `journalAbbreviation`. (as a reminder, the CSL JSON schema is based on Frank's design)



> If only container-title-short is officially a variable, would it make sense to prefer that, and fall back to journalAbbreviation only if it's not present?

Yes. `container-title-short` and `title-short` already exist in csl-data.json of CSL 1.0.1, and it's much clearer to have a 1:1 mapping between CSL JSON fields and CSL variables. We can officially mark `shortTitle` and `journalAbbreviation` as legacy fields and remove them in a future CSL release, and use them as fall-back fields for now in the CSL processors.

**Comment by fbennett** [comment]
Fixed in tagged release 1.1.211.
